qiskit/wrapper/_wrapper.py

```diff
@@ -21,35 +27,159 @@
+    Internal function that calls the actual providers.
     Args:
-            token (str): The token used to register on the online backend such
-                as the quantum experience.
-            url (str): The url used for online backend such as the quantum
-                experience.
-            hub (str): The hub used for online backend.
-            group (str): The group used for online backend.
-            project (str): The project used for online backend.
-            proxies (dict): Proxy configuration for the API, as a dict with
-                'urls' and credential keys.
-            verify (bool): If False, ignores SSL certificates errors.
-            provider_name (str): the unique name for the online backend
-                provider (for example, 'ibmq' for the IBM Quantum Experience).
+        token (str): The token used to register on the online backend such
+            as the quantum experience.
+        url (str): The url used for online backend such as the quantum
+            experience.
+        hub (str): The hub used for online backend.
+        group (str): The group used for online backend.
+        project (str): The project used for online backend.
+        proxies (dict): Proxy configuration for the API, as a dict with
+            'urls' and credential keys.
+        verify (bool): If False, ignores SSL certificates errors.
+        provider_name (str): the unique name for the online backend
+            provider (for example, 'ibmq' for the IBM Quantum Experience).
     Raises:
         QISKitError: if the provider name is not recognized.
     """
-    if provider_name == 'ibmq':
-        provider = IBMQProvider(token, url,
-                                hub, group, project, proxies, verify)
-        _DEFAULT_PROVIDER.add_provider(provider)
+    _registered_names = [p.name for p in _DEFAULT_PROVIDER.providers]
+    if provider_name not in _registered_names:
+        try:
+            provider = IBMQProvider(token, url,
+                                    hub, group, project,
+                                    proxies, verify,
+                                    provider_name)
+            _DEFAULT_PROVIDER.add_provider(provider)
+        except ConnectionError:
+            warnings.warn('%s not registered. No connection established.' % provider_name)
     else:
-        raise QISKitError('provider name %s is not recognized' % provider_name)
+        if rc_set.REGISTER_CALLED:
+            warnings.warn(
+                "%s already registered. Use unregister('%s') to remove." % (
+                    provider_name, provider_name))
+
 
+def unregister(provider_name):
+    """
+    Removes a provider of list of registered providers.
+    Args:
+        provider_name (str): The unique name for the online provider.
+
+    Raises:
+        QISKitError: if the provider name is not recognized.
+    """
+    if provider_name == 'core':
+        raise QISKitError("Cannot unregister 'core' provider.")
```

Comment:
Very random thought: why do we want to restrict unregistering the `core` provider?

Reply:
Because we can not register it, and what would be the use case to remove the 'core' solvers?

Reply:
:+1: